internal/provider/organization_resource.go

```diff
@@ -0,0 +1,229 @@
+		Description: data.Description.ValueString(),
+		Icon:        data.Icon.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create organization", err.Error())
+		return
+	}
+	tflog.Trace(ctx, "successfully created organization", map[string]any{
+		"id": org.ID,
+	})
+	// Fill in `ID` since it must be "computed".
+	data.ID = UUIDValue(org.ID)
+	// We also fill in  `DisplayName`, since it's optional but the backend will
+	// default it.
+	data.DisplayName = types.StringValue(org.DisplayName)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *OrganizationResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Read Terraform plan data into the model
+	var data OrganizationResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	orgID := data.ID.ValueUUID()
+
+	// Update the organization metadata
+	tflog.Trace(ctx, "updating organization", map[string]any{
+		"id":               orgID,
+		"new_name":         data.Name,
+		"new_display_name": data.DisplayName,
+		"new_description":  data.Description,
+		"new_icon":         data.Icon,
+	})
+	_, err := r.Client.UpdateOrganization(ctx, orgID.String(), codersdk.UpdateOrganizationRequest{
+		Name:        data.Name.ValueString(),
+		DisplayName: data.DisplayName.ValueString(),
+		Description: data.Description.ValueStringPointer(),
+		Icon:        data.Icon.ValueStringPointer(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update organization %s, got error: %s", orgID, err))
+		return
+	}
+	tflog.Trace(ctx, "successfully updated organization")
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *OrganizationResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Read Terraform prior state data into the model
+	var data OrganizationResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	orgID := data.ID.ValueUUID()
+
+	tflog.Trace(ctx, "deleting organization", map[string]any{
+		"id": orgID,
+	})
+	err := r.Client.DeleteOrganization(ctx, orgID.String())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete organization %s, got error: %s", orgID, err))
+		return
+	}
+	tflog.Trace(ctx, "successfully deleted organization")
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+}
+
+func (r *OrganizationResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Terraform will eventually `Read` in the rest of the fields after we have
+	// set the `id` attribute.
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
```

Comment:
We should also support importing by organization name.

Reply:
what if we actually _only_ supported importing by name?

Reply:
Is there any reason we wouldn't want to support both? Only supporting name would make it inconsistent with our other resources - is there any benefit?

Reply:
Whilst not quite the same, a lot of the resources in the GCP provider accept IDs in a bunch of different forms. 
https://registry.terraform.io/providers/hashicorp/google/6.11.1/docs/resources/gkeonprem_bare_metal_admin_cluster

Perhaps the lack of `x or y` imports is just because it's rare to have two totally unique identifiers for the same object on a SaaS.